src/message/messageActionSheet.js

```diff
@@ -226,30 +226,38 @@ const cancel = params => {};
 cancel.title = 'Cancel';
 cancel.errorMessage = 'Failed to hide menu';
 
-export const constructHeaderActionButtons = ({
-  backgroundData: { mute, subscriptions, ownUser },
-  stream,
+export const constructTopicActionButtons = ({
+  backgroundData: { mute, ownUser, streams, subscriptions, unreadStreams },
+  streamId,
   topic,
 }: {|
   backgroundData: $ReadOnly<{
     mute: MuteState,
+    streams: Map<number, Stream>,
     subscriptions: Subscription[],
+    unreadStreams: UnreadStreamsState,
     ownUser: User,
     ...
   }>,
-  stream: string,
+  streamId: number,
   topic: string,
-|}): Button<HeaderArgs>[] => {
+|}): Button<TopicArgs>[] => {
   const buttons = [];
   if (ownUser.is_admin) {
     buttons.push(deleteTopic);
   }
-  if (isTopicMuted(stream, topic, mute)) {
+  const unreadCount = unreadStreams.get(streamId)?.get(topic)?.size;
+  if (unreadCount !== undefined && unreadCount > 0) {
```

Comment:
Hmm interesting.

This is reasonable, but I'd kind of like to keep this kind of detail of the unreads data structure encapsulated inside the unreads model. I'm thinking perhaps a function in `unreadModel.js` that takes an `UnreadState` and a stream and topic, and returns a count.

It looks like we already have one other spot outside the unreads code that similarly looks inside this data structure -- in `topicSelectors.js`:
```js
      const unreadCount = unreadStreams.get(stream.stream_id)?.get(name)?.size ?? 0;
```
(I found that in a kind of crude, but effective, way: by editing the definition of `UnreadStreamsState` to say `export opaque type UnreadStreamsState = …`, and looking through the resulting Flow errors to find any outside the unreads code.)

I'll write up a quick change to factor that `topicSelectors.js` line out into the function I have in mind, and push that to this PR branch. Then you can rebase it (after fetching the new version of your branch) to come before this commit, and call that function instead.

Reply:
> I'll write up a quick change to factor that `topicSelectors.js` line out into the function I have in mind, and push that to this PR branch.

OK, that change grew a bit as I wrote it, so I sent it as its own PR: #4805 .

If this PR is otherwise ready before that one is merged, no need to wait for it; we can refactor this to use that as a followup. But that PR is pretty short and simple, so it's likely we'll have it merged soon anyway.